setup.cfg

```
@@ -149,8 +149,7 @@ install_requires =
     pyyaml>=5.1
     rich>=9.2.0
     setproctitle>=1.1.8, <2
-    # SQLAlchemy 1.4 breaks sqlalchemy-utils https://github.com/kvesteri/sqlalchemy-utils/issues/505
-    sqlalchemy>=1.3.18, <1.4
+    sqlalchemy>=1.3.18
```

WDYT about bumping it to 1.3.24 ? last version in 1.3 series?

```suggestion
    sqlalchemy>=1.3.24
```

---

I think lower bounds are pretty meaningless unless we know we need some feature. They are really only used if someone has a version below that - in which case it will automatically upgrade the dependency when they upgrade airflow. I ten'd to leave the lower bounds unchanged unless there is a good reason to do so.

---

If someone has 1.3.19 installed, this won't bump it up as it satisfies the version. Like you said not really needed, but hey a good number of bug fixes between 1.3.18 and 1.3.24: https://docs.sqlalchemy.org/en/14/changelog/changelog_13.html

and we already use 1.3.24 in constraints :)

---

I'd generally say don't bump minimum unless we know we need that fix.

As a library we want to be as permissive as we can with the _requirements_.

---

Actually nevermind, lower has an advantage to cause less conflicts with other dependencies that depend on an older version of `sqlalchemy`

---

Exactly that :)

---

I don’t have a preference, in practice the difference is minimal. The only behavioural difference I can think of is if the user `pip install --upgrade airflow` without specifying the constraints, but from I know most users don’t upgrade an existing setup anyway...